src/main/java/buildcraft/factory/BlockTank.java

```diff
@@ -88,123 +88,210 @@ public IIcon getIconAbsolute(IBlockAccess iblockaccess, int i, int j, int k, int
-
-            if (tile instanceof TileTank) {
-                TileTank tank = (TileTank) tile;
-                if (tank.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid != null) {
-                    entityplayer.addChatComponentMessage(
-                            new ChatComponentText(
-                                    "Amount: " + tank.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid.amount + " mB"));
+            return false;
+        }
+
+        // IFluidContainerItem (IFCI) Logic (Large Fluid Cells)
+        if (!(held.getItem() instanceof IFluidContainerItem)) return false;
+        if (world.isRemote) return true; // server does transfers
+
+        final IFluidContainerItem fluidContainer = (IFluidContainerItem) held.getItem();
+        final boolean sneaking = player.isSneaking();
+        final int INF_MB = Integer.MAX_VALUE / 4;
+
+        // Support for stacked cells
+        ItemStack work = held.copy();
+        work.stackSize = 1;
+
+        // prioritizes the bottom for tank stacks
+        TileTank bottom = clicked.getBottomTank();
+        if (bottom == null) return true;
+
+        int moved = 0;
+
+        // Push from cell -> tank
+        if (!sneaking) {
+            FluidStack containerFluid = fluidContainer.getFluid(work);
+            if (containerFluid != null && containerFluid.amount > 0) {
+                // fill the tank; this returns how much it actually accepted
+                moved = bottom.fill(ForgeDirection.UNKNOWN, containerFluid, true);
+
+                // in creative mode we do NOT drain the cell; it stays full
+                if (moved > 0 && !creative) {
+                    fluidContainer.drain(work, moved, true);
                 }
             }
         }
 
-        return false;
+        // Pull from tank -> cell (stack friendly)
+        if (sneaking) {
+            FluidStack inCell = fluidContainer.getFluid(work);
+            FluidStack stackFluid = bottom.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid;
+            net.minecraftforge.fluids.Fluid target = (inCell != null && inCell.amount > 0) ? inCell.getFluid()
+                    : (stackFluid != null ? stackFluid.getFluid() : null);
+
+            if (target != null) {
+                int accept = fluidContainer.fill(work, new FluidStack(target, INF_MB), false);
+                if (accept > 0) {
+                    // tank stack protection consistency
+                    FluidStack drained = bottom.drain(ForgeDirection.UNKNOWN, new FluidStack(target, accept), true);
+                    if (drained != null && drained.amount > 0) {
+                        if (creative) {
+                            moved = drained.amount; // tank changed, item remains
+                        } else {
+                            int filled = fluidContainer.fill(work, drained, true);
+                            moved = filled;
+
+                            // remainder logic
+                            if (filled < drained.amount) {
```

Comment:
not sure if this is ever possible - you've just ran the simulation (which has the same checks as the real filling/draining) that said you can fill `accept` mB of fluid, then you tried to fill `drained` mB of fluid, where `drained <= accept`, so `filled` always equals `drained`